agntcy_acp/langgraph/acp_node.py

```diff
@@ -192,27 +199,114 @@ def _handle_run_output(self, state: Any, run_output: RunOutput):
             run_error: RunError = run_output.actual_instance
             raise ACPRunException(f"Run Failed: {run_error}")
         elif isinstance(run_output.actual_instance, RunInterrupt):
-            raise ACPRunException(f"ACP Server returned a unsupporteed interrupt response: {run_output}")
+            # This case is handled in the invokes
+            pass
+
         else:
-            raise ACPRunException(f"ACP Server returned a unsupporteed response: {run_output}")
+            raise ACPRunException(
+                f"ACP Server returned a unsupporteed response: {run_output}"
+            )
 
         return state
 
-    def invoke(self, state: Any, config: RunnableConfig) -> Any:
+    def _get_thread_id(self, config):
+        configurable = config.get("configurable", {})
+        thread_id = configurable.get("thread_id", None)
+        return thread_id
+
+    def _handle_interrupt(self, config, run_output):
+        thread_id = self._get_thread_id(config=config)
+        value = run_output.output.actual_instance.interrupt["default"]
+
+        if (
+            thread_id is not None
+            and self.previous_thread_run.get(thread_id, None) is None
+        ):
+            self.previous_thread_run[thread_id] = run_output
+
+        interrupt_result = interrupt(value)
+        if thread_id is not None and thread_id in self.previous_thread_run:
+            del self.previous_thread_run[thread_id]
+
+        return interrupt_result
+
+    def invoke(
+        self,
+        state: dict[str, Any] | Any,
+        config: RunnableConfig | None = None,
+        **kwargs: Any,
+    ) -> Any:
+
+        thread_id = self._get_thread_id(config)
         run_create = self._prepare_run_create(state, config)
```

Comment:
You don't need a run_create in case of interrupt resumption, so you could call the _prepare_run_create only when you are sure that you need it.